pydantic/_internal/_validate_call.py

```diff
@@ -76,11 +70,9 @@ def __init__(self, function: Callable[..., Any], config: ConfigDict | None, vali
                 else Any
             )
             gen_schema = _generate_schema.GenerateSchema(config_wrapper, namespace)
-            schema = gen_schema.collect_definitions(gen_schema.generate_schema(return_type))
-            schema = _discriminated_union.apply_discriminators(simplify_schema_references(schema))
+            schema = gen_schema.clean_schema(gen_schema.generate_schema(return_type))
             self.__return_pydantic_core_schema__ = schema
             core_config = config_wrapper.core_config(self)
-            schema = validate_core_schema(schema)
             validator = pydantic_core.SchemaValidator(schema, core_config)
```

Comment:
(it's possible it should be using `create_schema_validator` but maybe it wasn't added on purpose because that is only intended to be run when a user defines a model, not this sort of dynamically generated anonymous model). @samuelcolvin any thoughts?

Reply:
I mean, why use `SchemaValidator(schema, core_config)` instead of `create_schema_validator(schema, core_config, config_wrapper.plugin_settings)`? In other words, why avoid using plugins here?

Reply:
> My point above is that I'm not sure if it was an unintended omission or an intentional choice.

Gotcha, we were commenting at the same time and I wasn't sure if my initial comment had been clear.

> maybe it wasn't added on purpose because that is only intended to be run when a user defines a model, not this sort of dynamically generated anonymous model

Hmmm, but `create_schema_validator` *is* being used in the same method just a few lines up.

On a related note, there doesn't seem to be any reason to create another `core_config`.

Reply:
> Hmmm, but create_schema_validator is being used in the same method just a few lines up.

That pretty much confirms that it was an omission, please go ahead and add it.

> On a related note, there doesn't seem to be any reason to create another core_config

I'm not sure what you mean specifically but if we're creating a new `core_config` from a `ConfigWrapper` when we could have just referenced an existing one please fix that.